server/server.go

```diff
@@ -0,0 +1,83 @@
+package server
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/skuid/keyman/sign"
+	"go.uber.org/zap"
+	"golang.org/x/crypto/ssh"
+)
+
+type KeyRequest struct {
+	Key        string   `json:"key"`
+	Principals []string `json:"principals"`
+}
+
+type KeyResponse struct {
+	Certificate string `json:"certificate"`
+}
+
+type Server struct {
+	CA sign.Signer
+}
+
+func (s *Server) SignHandler(w http.ResponseWriter, r *http.Request) {
+
+	kr := &KeyRequest{}
+
+	err := json.NewDecoder(r.Body).Decode(kr)
+	if err != nil {
+		http.Error(w, "Could not decode input", http.StatusBadRequest)
+		zap.L().Debug("Could not decode input", zap.Error(err))
+		return
+	}
+
+	// TODO change username, parameterize duration
+	signedCert, err := s.CA.Sign(
+		kr.Key,
+		"micahhausler",
+		kr.Principals,
+		(time.Duration(8) * time.Hour),
+	)
+	if err != nil {
+		http.Error(w, "Error signing request", http.StatusInternalServerError)
+		zap.L().Error("Error signing request", zap.Error(err))
+		return
+	}
+
+	err = json.NewEncoder(w).Encode(&KeyResponse{Certificate: signedCert})
```

Comment:
`is err := json.NewEncoder(w).Encode(&KeyResponse{Certificate: signedCert}); err != nil {...}`